### ISSUE TITLE
executeTransaction fails when used with ArangoDB 3.6

### ISSUE THREAD

**Issue by mrwatson** [arangodb/arangojs#644]
After upgrading to ArangoDB 3.6, our existing `executeTransaction` calls fail with the error:

```
ArangoError: 'collection' is missing, expecting /_api/document POST /_api/document/<collection> or query parameter 'collection'
      at new ArangoError (node_modules/arangojs/src/error.ts:76:17)
      at Object.resolve (node_modules/arangojs/src/connection.ts:369:20)
      at callback (node_modules/arangojs/src/connection.ts:211:16)
      at IncomingMessage.<anonymous> (node_modules/arangojs/src/util/request.node.ts:117:15)
```

This error is consistent across all uses of `executeTransaction`. Sample code for one instance is below:
```
    const action = String(
      /* istanbul ignore next */
      function(params) {
        // eslint-disable-next-line global-require
        const { query, db } = require('@arangodb')

        // Check to see if any documents already exist.
        const existingDocs = query`
      FOR doc, link IN 1..1 OUTBOUND ${params.linkSourceId} ${params.edgeCollectionName}
        FILTER link.type == ${params.linkType}
        FILTER doc[${params.fieldName}] == ${params.fieldValue}
        RETURN doc
    `.toArray()

        if (existingDocs.length === 0) {
          // No documents exist, so create one.
          const collection = db._collection(params.collectionName)
          return collection.save(params.doc)
        } else {
          // Found existing document(s) - return null.
          return null
        }
      }
    )

    return this.arangodb.executeTransaction(collectionName, action, {
      params: {
        edgeCollectionName: EDGES_COLLECTION,
        collectionName,
        linkSourceId,
        doc: document,
        fieldName,
        fieldValue,
        linkType
      }
    })
```

This project is using arangojs 6.12.0 and ArangoDB 3.6.0.

**Comment by hkernbach** [comment]
Is your `collectionName` filled? Or maybe empty/undefined? 
As I could run your query with: 

```
  let collectionName = "persons";
  let linkSourceId = "132";
  let EDGES_COLLECTION = "edgecollection";
  let document = {};
  let fieldName = "peter";
  let fieldValue = "many";
  let linkType = "oo0";

  const action = String(
    /* istanbul ignore next */
    function(params) {
      // eslint-disable-next-line global-require
      const { query, db } = require('@arangodb')

      // Check to see if any documents already exist.
      const existingDocs = query`
    FOR doc, link IN 1..1 OUTBOUND ${params.linkSourceId} ${params.edgeCollectionName}
      FILTER link.type == ${params.linkType}
      FILTER doc[${params.fieldName}] == ${params.fieldValue}
      RETURN doc
  `.toArray()

      if (existingDocs.length === 0) {
        // No documents exist, so create one.
        const collection = db._collection(params.collectionName)
        return collection.save(params.doc)
      } else {
        // Found existing document(s) - return null.
        return null
      }
    }
  )

  return db.executeTransaction(collectionName, action, {
    params: {
      edgeCollectionName: EDGES_COLLECTION,
      collectionName,
      linkSourceId,
      doc: document,
      fieldName,
      fieldValue,
      linkType
    }
  })
```

**Comment by pluma** [comment]
The error message indicates `collectionName` is empty. Can you dump the values of the variables you're passing to `executeTransaction` for me?

**Comment by pluma** [comment]
@mrwatson Upon closer inspection it looks like the error you're seeing is actually from a different part of your code. The error indicates this is from the `/_api/document` API, not the `/_api/transaction` route.

Do you have any code inserting a document with arangojs maybe?

**Comment by pluma** [comment]
If this is any help, I was able to reproduce the error message using the following code:

```
await db.collection("").save(/* anything here but can also be empty */)
```

Could it be that you somehow have an empty collection name somewhere?

**Comment by pluma** [comment]
I'm closing this issue due to a lack of activity. If the isue persists, please feel free to reopen with additional information.

**Comment by mrwatson** [comment]
Okay. Sorry I forgot to post the resolution here, but we actually resolved the issue by rolling back to ArangoDB 3.5.4.
